app/services/post_importer.rb

```diff
@@ -0,0 +1,69 @@
+class PostImporter < Object
+  def initialize(url)
+    @url = url
+  end
+
+  def import(board_id, section_id, status, threaded, importer_id)
+    validate_url!
+    validate_duplicate!(board_id) unless threaded
+    validate_usernames!
+
+    ScrapePostJob.perform_later(@url, board_id, section_id, status, threaded, importer_id)
+  end
+
+  private
+
+  def validate_url!
+    raise InvalidDreamwidthURL.new('Invalid URL provided.') unless self.class.valid_dreamwidth_url?(@url)
+  end
+
+  def validate_duplicate!(board_id)
+    subject = dreamwidth_doc.at_css('.entry .entry-title').text.strip
+    subj_post = Post.where(subject: subject, board_id: board_id).first
+    return unless subj_post
+    raise AlreadyImported.new("This thread has already been imported! " + ScrapePostJob.view_post(subj_post.id))
+  end
```

Comment:
Given this code is identical to https://github.com/Marri/glowfic/blob/9389578a83fd7b2056d59920669e31189be165b8/app/services/post_scraper.rb#L162-L166, should we try to deduplicate this somehow?